src/main/java/com/google/api/codegen/discovery/MainDiscoveryProviderFactory.java

```diff
@@ -49,48 +49,61 @@
   public static final String PYTHON = "python";
   public static final String RUBY = "ruby";
 
-  private static final String DEFAULT_SNIPPET_FILE = "discovery_fragment.snip";
+  private static final String SNIPPET_FILE_ADC = "discovery_fragment.snip";
+  private static final String SNIPPET_FILE_3LO = "discovery_fragment_3lo.snip";
+  private static final String SNIPPET_FILE_API_KEY = "discovery_fragment_api_key.snip";
 
   public static DiscoveryProvider defaultCreate(
       Service service, ApiaryConfig apiaryConfig, String id) {
+    String snippetFile;
+    // TODO(saicheems): Update this as new auth types are supported.
+    switch (apiaryConfig.getAuthType()) {
+      case APPLICATION_DEFAULT_CREDENTIALS:
+      case OAUTH_3L:
+      case API_KEY:
+        snippetFile = SNIPPET_FILE_ADC;
+        break;
+      default:
+        throw new RuntimeException("unsupported auth type");
+    }
```

Comment:
Hmm, but then you'd be giving `AuthType` an additional abstraction. If we do this, it would be better renamed `AuthTemplate`.